soot-infoflow-android/src/soot/jimple/infoflow/android/entryPointCreators/AndroidEntryPointCreator.java

```diff
@@ -326,6 +338,60 @@ protected SootMethod createDummyMainInternal() {
 		return mainMethod;
 	}
 
+	private void createJavascriptCallbacks() {
+		Jimple j = Jimple.v();
+		Hierarchy h = Scene.v().getActiveHierarchy();
+		for (SootMethod m : javascriptInterfaceStmts.keySet()) {
+			Set<Stmt> statements = javascriptInterfaceStmts.get(m);
+			for (Stmt s : statements) {
+				UnitPatchingChain units = m.retrieveActiveBody().getUnits();
+				Stmt succ = (Stmt) units.getSuccOf(s);
+				SootField f = null;
+				Value arg = s.getInvokeExpr().getArg(0);
+				if (succ.getTag(SimulatedCodeElementTag.TAG_NAME) != null) {
+					if (succ instanceof AssignStmt) {
+						AssignStmt assign = (AssignStmt) succ;
+						if (assign.getRightOp() == arg && assign.getLeftOp() instanceof StaticFieldRef) {
+							f = ((StaticFieldRef) assign.getLeftOp()).getField();
+						}
+					}
+				}
+				if (f == null) {
+					//create field
+					f = createField(arg.getType(), "jsInterface");
+					AssignStmt assign = j.newAssignStmt(j.newStaticFieldRef(f.makeRef()), arg);
+					assign.addTag(SimulatedCodeElementTag.TAG);
+					units.insertAfter(assign, s);
+				}
+
+				Local l = j.newLocal(f.getName(), f.getType());
+				body.getLocals().add(l);
+				Stmt assignF = j.newAssignStmt(l, j.newStaticFieldRef(f.makeRef()));
+				body.getUnits().add(assignF);
+				SootClass cbtype = ((RefType) f.getType()).getSootClass();
+				List<SootClass> allPossibleImpls;
+				if (cbtype.isInterface()) {
```

Comment:
So einfach ist das nicht. Du kannst auch eine Subklasse haben, die ein Interface implement, das eine Default-Methode hat. Haben wir für so etwas nicht längst eine Hilfsmethode, z.B. in den TypeUtils? Ich erinnere mich dunkel, das mal implementiert zu haben.